Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -1138,6 +1148,60 @@ namespace Tensile
         return rv;
     }
 
+    bool ContractionSolution::getMatrixInstructionFromKernelName(
+        vector4<std::uint32_t>& matrixInst) const
+    {
+        std::string regexp_string(KERNEL_NAME_MIINST_REGEX);
+        std::regex  miRegex(regexp_string);
+        std::smatch matches;
+
+        std::string kName = this->KernelName();
+
+        if(std::regex_search(kName, matches, miRegex))
+        {
+            if(matches.size() != KERNEL_NAME_MIINST_REGEX_NUM_MATCHES)
+            {
+                return false;
+            }
+            // the first sub_match element (index 0) corresponds to the entire match
```

Comment:
Maybe it is shorter this way:
 bool ret = std::regex_search(kName, matches, miRegex) && matches.size() == KERNEL_NAME_MIINST_REGEX_NUM_MATCHES;
        if(ret)
        {
            // the first sub_match element (index 0) corresponds to the entire match
            matrixInst.x = atoi(matches[1].str().c_str());
            matrixInst.y = atoi(matches[2].str().c_str());
            matrixInst.z = atoi(matches[3].str().c_str());
            matrixInst.w = atoi(matches[4].str().c_str());
        }

        return ret;